x/ecocredit/server/msg_server.go

```diff
@@ -345,5 +367,50 @@ func retire(ctx types.Context, store sdk.KVStore, recipient string, batchDenom b
 		Retirer:    recipient,
 		BatchDenom: string(batchDenom),
 		Units:      math.DecimalString(retired),
+		Location:   location,
 	})
 }
+
+// validateLocation checks that the country and region conform to ISO 3166 and
```

Comment:
what about extracting all the location validation related functions into a separate file? and then have a dedicated test file for it, eg location_test.go instead of msg_server_test.go